legion/main.cc

```diff
@@ -107,8 +107,10 @@ void leaf(const Task *task,
     }
   }
 
+  char *scratch_ptr = (char *) malloc(graph.kernel.kernel_arg.scratch_bytes_per_task);
```

Comment:
Let's make this a region with a single disjoint partition. We don't want to be calling malloc during the run.

Reply:
Thanks, fixed.